[c10d] fix broken test

c-p-i-o opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137375

Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

c-p-i-o committed:
[c10d] fix broken test

Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

fduwjj commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -4287,9 +4286,10 @@ def test_timeout_dumps_on_stuck_ranks(self):
         os.environ["TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC"] = "1"
         # restore this env var to its prior default in case another test changed it
         os.environ["TORCH_NCCL_COORD_CHECK_MILSEC"] = "1000"
+        os.environ["TORCH_NCCL_ASYNC_ERROR_HANDLING"] = "0"
```
this is interesting because the default value in prod is 3, so we should still keep it as 3 to make sure this test pass?

wconstab replied:
can you explain the reasoning for 'NoHandling' (mode=0) fixing this test?

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

c-p-i-o committed:
Update on "[c10d] fix broken test"


Summary:
TORCH_NCCL_ASYNC_ERROR_HANDLING immediately aborts rank0 without giving
it time to write out flight recorder files.
This was causing test to fail.

Test Plan:
python test/distributed/test_c10d_nccl.py -k
NCCLTraceTestTimeoutDumpOnStuckRanks.test_timeout_dumps_on_stuck_ranks

Reviewers:

Subscribers:

Tasks:

Tags:

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]
